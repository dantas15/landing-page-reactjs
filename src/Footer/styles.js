import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
