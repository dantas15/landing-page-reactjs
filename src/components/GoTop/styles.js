import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;

    bottom: 2rem;
    right: 2rem;

    z-index: 6;

    border-radius: 0.4rem;

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `}
`;
