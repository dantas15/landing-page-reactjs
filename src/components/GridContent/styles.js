import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
