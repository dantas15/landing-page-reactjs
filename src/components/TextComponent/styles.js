import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`;
