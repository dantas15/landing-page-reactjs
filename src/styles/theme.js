export const theme = {
  colors: {
    mainBg: 'red',
    secondaryBg: 'yellow',
  },
  fonts: {},
  spacings: {},
};
