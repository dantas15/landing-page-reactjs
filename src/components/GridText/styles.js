import styled, { css } from 'styled-components';

import { Title as HeadingContainer } from 'components/Heading/styles';
import { Container as TextComponent } from 'components/TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;

    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xhuge};
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    } ;
  `}
`;
