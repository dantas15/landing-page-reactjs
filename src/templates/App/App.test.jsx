import { screen } from '@testing-library/dom';

import { Home } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

test('renders home', () => {
  const { debug } = renderTheme(<Home />);
  debug();

  const headingContainer = screen.getByRole('heading', {
    name: 'Hello world',
  }).parentElement;
  expect(headingContainer).toHaveStyle({
    background: theme.colors.mainBg,
  });
  expect(headingContainer).toMatchSnapshot();
  expect(headingContainer).toHaveStyleRule('background', 'red');
});
