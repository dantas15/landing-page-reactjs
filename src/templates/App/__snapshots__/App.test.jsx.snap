// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home 1`] = `
.c0 {
  background: red;
}

<div
  class="c0"
>
  <h1>
    Hello world
  </h1>
</div>
`;
